### PR TITLE
[WIP] Predict Loss as Output

### PR DESCRIPTION
**Pull request by lilleswing**
Allows predicting the loss as output
```python
import numpy as np
import deepchem as dc

# Only for debug!
np.random.seed(123)

# Load Tox21 dataset
n_features = 1024
tox21_tasks, tox21_datasets, transformers = dc.molnet.load_tox21()
train_dataset, valid_dataset, test_dataset = tox21_datasets

# Fit models
metric = dc.metrics.Metric(dc.metrics.roc_auc_score, np.mean)

model = dc.models.MultiTaskClassifier(
    len(tox21_tasks),
    n_features,
    layer_sizes=[1000],
    dropouts=[.25],
    learning_rate=0.001,
    batch_size=50,
    use_queue=False)

# Fit trained model
model.fit(train_dataset, nb_epoch=1)
valid_loss = model.predict(valid_dataset, outputs=model.loss)
print(np.mean(valid_loss))
```

Bug is brought up here https://github.com/deepchem/deepchem/issues/1040

### CODE REVIEW

File: deepchem/models/tensorgraph/tensor_graph.py [hunk 1]
```diff
@@ -279,11 +279,11 @@ def default_generator(self,
           deterministic=deterministic,
           pad_batches=pad_batches):
         feed_dict = dict()
-        if len(self.labels) == 1 and y_b is not None and not predict:
+        if len(self.labels) == 1 and y_b is not None:
```
Comment (lilleswing) [inline deepchem/models/tensorgraph/tensor_graph.py]: So I initially added the predict flag in April.  
Back then we did not allow arbitrary outputs of predict.

We already have a None check for the generator.  Is there any reason besides performance to not always pass in the tensor?  If a user is a power user and wants performance they can create a dataset without None values

File: deepchem/models/tensorgraph/tensor_graph.py [hunk 2]
```diff
@@ -349,7 +349,11 @@ def predict_on_generator(self, generator, transformers=[], outputs=None):
 
       final_results = []
       for result_list in results:
-        final_results.append(np.concatenate(result_list, axis=0))
+        try:
```
Comment (lilleswing) [inline deepchem/models/tensorgraph/tensor_graph.py]: Since loss is an aggregated statistic it can't be concatenated (only a single entry per mini-batch).

We shouldn't fail hard in these cases -- but instead just return the results to the user chunked into mini-batches.

**Comment by lilleswing** [comment]
Still need to add tests -- just wanted to see what the CI system says.

**Comment by fuzunyun** [comment]
default
default

**Comment by fuzunyun** [comment]
I tried your code but some error occured which I have no idea how it happened. Could you please help me find out where the problem is?  Attached is my full script.
Thanks again for your help.
Regards,
default
default
